src/Evolution/DgSubcell/Actions/SelectNumericalMethod.hpp

```diff
@@ -0,0 +1,67 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <tuple>
+#include <utility>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Evolution/DgSubcell/Actions/Labels.hpp"
+#include "Evolution/DgSubcell/ActiveGrid.hpp"
+#include "Evolution/DgSubcell/Tags/ActiveGrid.hpp"
+#include "Parallel/Actions/Goto.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace Parallel {
+template <typename Metavariables>
+class GlobalCache;
+}  // namespace Parallel
+namespace tuples {
+template <typename...>
+class TaggedTuple;
+}  // namespace tuples
+/// \endcond
+
+namespace evolution::dg::subcell::Actions {
+/*!
+ * \brief Goes to `Labels::BeginDg` or `Labels::BeginSubcell` depending on
+ * whether the active grid is `Dg` or `Subcell`.
+ *
+ * GlobalCache:
+ *
+ * DataBox:
+ * - Uses:
+ *   - `subcell::Tags::ActiveGrid`
+ */
+struct SelectNumericalMethod {
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static std::tuple<db::DataBox<DbTagsList>&&, bool, size_t> apply(
+      db::DataBox<DbTagsList>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      const Parallel::GlobalCache<Metavariables>& /*cache*/,
+      const ArrayIndex& /*array_index*/, ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) noexcept {
+    if (db::get<Tags::ActiveGrid>(box) == subcell::ActiveGrid::Dg) {
+      const size_t dg_index =
+          tmpl::index_of<ActionList, ::Actions::Label<Labels::BeginDg>>::value +
+          1;
```

Comment:
For my understanding:

The dox indicate this "goes to" `Labels::BeginDg`, so why are you returning an index one past that? Is the "+1" in the code wrong, or should the dox say "point the algorithm to the action after Labels::BeginDg", or is my understanding of the infrastructure mistaken?

(Same question for subcell below)

Reply:
Right, the `+1` in the code is so that we don't evaluate the label action, since that doesn't do anything anyway, so from a practical standpoint, going to `Label` or `Label+1` will have the same effect on control flow, just with `Label+1` you avoid calling the empty label action. I omitted this from the documentation since I figured the documentation is more on the "what are the side effects/DataBox changes that will occur" and since empty actions do nothing (other than maybe use up a handful of CPU cycles) I didn't bother noting this. If you think it's important to note, I'll add it :)